libcore/src/geometry/Building.cpp

```diff
@@ -1252,4 +1187,100 @@ std::vector<TrainType> Building::GetTrainTypes()
     return trainTypes;
 }
 
+void Building::AddTrainAddedWall(int trainID, Wall trainAddedWall)
+{
+    auto iter = _trainAddedWalls.find(trainID);
+
+    if(iter != _trainAddedWalls.end()) {
+        iter->second.emplace_back(trainAddedWall);
+    } else {
+        _trainAddedWalls.emplace(trainID, std::vector<Wall>{trainAddedWall});
+    }
+}
+
+void Building::SetTrainAddedWalls(int trainID, std::vector<Wall> trainAddedWalls)
+{
+    _trainAddedWalls[trainID] = trainAddedWalls;
+}
+
+std::optional<std::vector<Wall>> Building::GetTrainAddedWalls(int trainID)
```

Comment:
Can we name this `GetTrainWallsAdded` to sound more like the walls added from the train? Or do i misunderstand?

Reply:
Changed, also changed the names of the other functions to fit the naming